src/app/shell/dimAngularFilters.filter.js

```diff
@@ -166,11 +166,16 @@ mod.filter('sortItems', (dimSettingsService) => {
     }
 
     items = _.sortBy(items || [], 'name');
-    if (sort === 'primaryStat' || sort === 'rarityThenPrimary' || sort === 'quality' || sort === 'typeThenPrimary') {
+    if (sort === 'primaryStat' || sort === 'rarityThenPrimary' || sort === 'quality' || sort === 'typeThenPrimary' || sort === 'basePowerThenPrimary') {
       items = _.sortBy(items, (item) => {
         return (item.primStat) ? (-1 * item.primStat.value) : 1000;
       });
     }
+    if (sort === 'basePowerThenPrimary') {
+      items = _.sortBy(items, (item) => {
+        return (item.basePower) ? (-1 * item.basePower) : 1000;
```

Comment:
Suppose something like this would keep d1 support:
`return (item.basePower) ? (-1 * item.basePower) : ((item.primStat) ? (-1 * item.primStat.value) : 1000);`

Reply:
I would guess we actually get that behavior already, as I've already sorted on primaryStat and the whole function relies on _.sortBy using a "stable" sorting algorithm? This is based on my assumption that basePower is null or 0 for all D1 items though, so I might be in the wrong here.

Reply:
oh, right! didn't notice that you added the `|| basePower` above, looks good to me then.

Reply:
Yeah, it's actually a feature on D2 as well, as it allows the modded items to be sorted before the unmodded items with same basePower. :)